### PR TITLE
Update block to deny and add implicit consent for sending messages

### PR DESCRIPTION
**Pull request by nplasterer**
This addresses the items outlined here: https://github.com/orgs/xmtp/discussions/49#discussioncomment-7429217

Changes the word blocked to denied and adds implicit consent for sending a message to a conversation with unknown consent.

### CODE REVIEW

File: example/src/HomeScreen.tsx [hunk 1]
```diff
@@ -87,10 +87,10 @@ function ConversationItem({ conversation, client }: { conversation: Conversation
             ({messages?.length} messages)
           </Text>
           <Button
-            title="Block"
-            onPress={blockContact}
+            title="Denied"
```
Comment (rygine) [inline example/src/HomeScreen.tsx]: should this be `Deny`?

File: src/lib/Conversation.ts [hunk 2]
```diff
@@ -114,7 +114,7 @@ export class Conversation {
     }
   }
 
-  async consentState(): Promise<"allowed" | "blocked" | "unknown"> {
+  async consentState(): Promise<"allowed" | "denied" | "unknown"> {
```
Comment (rygine) [inline src/lib/Conversation.ts:117]: i see this string union type defined twice (here and in `index.ts`). think it's worth it to create 1 type and reference that instead?

**Review by rygine — APPROVED** [review]
LGTM. left some comments, but approving to unblock.

File: src/lib/Conversation.ts (hunk 2 above)
Comment (nplasterer) [reply src/lib/Conversation.ts:117]: Good call.

**Pull request merged**